backend/canisters/local_user_index/impl/src/updates/install_bot.rs

```diff
@@ -40,4 +39,21 @@ async fn install_bot(args: Args) -> Response {
     Success
 }
 
+fn prepare(args: &Args, state: &RuntimeState) -> Result<UserId, Response> {
+    let caller = state.env.caller();
+    let user = state.data.global_users.get(&caller).unwrap();
+    let bot = state.data.bots.get(&args.bot_id).ok_or(Response::NotFound)?;
+
+    match bot.registration_status {
+        BotRegistrationStatus::Public => (),
+        BotRegistrationStatus::Private(location) => {
+            if !location.is_some_and(|loc| loc == args.location) || bot.owner_id == user.user_id {
```

Comment:
This is slightly wrong. I think if you're going to do it on one line then use
```suggestion
if location.is_none_or(|loc| loc != args.location) && bot.owner_id != user.user_id {
```